Create a CLI package to make it easier to install React Navigation packages (Fix issue: #8384)

react-one opened the pull request:
# Create a CLI package to make it easier to install React Navigation packages ( Fix Issue: #8384 )

Mainly for now, an `add` command that allows us to install a package, with auto-detection of what package manager is in use. And automatic installation of peer dependencies. And also auto adding of `import 'react-native-gesture-handler'`

- [x] Installing package by name by fetching metadata. And can precise version (ex: stack@6.0.0)
- [x] Display error if the package doesn’t exist

- [x] Pick up the right installer following the logic
- Pick user picked installer if --installer option is provided
- A check and pick expo (expo in package.json)
- If not check yarn.lock (pick yarn)
- If not check package.lock (pick npm)
- If not check if yarn is installed (pick yarn)
- If not pick npm

- [x] Installing peer dependencies read from the metadata.

- [x] Auto-add of import 'react-native-gesture-handler'
- - [x] Check import 'react-native-gesture-handler'. And checking for both index.js|ts and App.js|ts.
- - [x] Add it automatically if not already added with a confirmation prompt.
- - [x] Auto picks up import quote style from code. If none found default to a single quote.

<br>

Done too:

- [x] Using loader in cli (ora package) and confirmation prompt (enquirer package)

- [x] Implemented logging with levels (with --verbose and --debug cli options too)

- [x] Implemented tests for installer pick up. And for react-native-gesture-handle. Used memfs ( in-memory file system ) and fs mocking for tests.

- [x] initial doc ( doc file )

## Command line in action

satya164 commented on packages/cli/src/utils/logger.ts:
```diff
@@ -0,0 +1,79 @@
+export class Logger {
+  private _isVerbose = false;
+  private _isDebug = false;
+  private _isLogEnabled = true;
+
+  public setIsVerbose(enable: boolean): this {
```
Don't use `_` for private properties (it's used in JS as a convention as there's no private properties there), or `public` to match code conventions in the repo.

satya164 commented on packages/cli/package.json:
```diff
@@ -0,0 +1,31 @@
+    "shelljs": "^0.8.4"
+  },
+  "devDependencies": {
+    "@types/jest": "^27.0.2",
+    "@types/node": "^16.9.4",
+    "@types/shelljs": "^0.8.9",
+    "jest": "^27.2.1",
+    "memfs": "^3.3.0",
+    "ts-jest": "^27.0.5",
+    "ts-node": "^9.0.0",
+    "typescript": "^4.4.3"
```
We shouldn't duplicate deps such as `typescript` or `jest` here as they are in the root project already. Also `ts-jest` doesn't seem necessary here.

satya164 reviewed (changes requested):
I left some comments, but it's a sizable PR so I haven't completed the review yet.

satya164 commented:
thanks @react-one. can you post about the ux flow of the cli and features on our discord so others can provide feedback?

react-one commented:
- When executing commands using `shell.exec()`. The output wasn't colored. I fixed that by creating a custom exec function based on spawn. I used cross-spawn for cross-platform compatibility.
- And I profited and added colors for our logs.
